src/CppUTestExt/MockExpectedCall.cpp

```diff
@@ -279,7 +293,7 @@ bool MockCheckedExpectedCall::hasInputParameter(const MockNamedValue& parameter)
 bool MockCheckedExpectedCall::hasOutputParameter(const MockNamedValue& parameter)
 {
     MockNamedValue * p = outputParameters_->getValueByName(parameter.getName());
-    return (p) ? true : ignoreOtherParameters_;
+    return (p) ? p->compatibleForCopying(parameter) : ignoreOtherParameters_;
```

Comment:
Compare it with the sibling `hasInputParameter`. The latter checks that there is an expected call with the same name, same parameter type and same parameter value (these last two by using `equals`). In the case of ouput parameters, checking value is not necessary, but checking type is necessary, otherwise we could copy an incompatible type in the top of the other, so we need an extra function that checks that types check.

In the old implementation that wasn't necessary, because the types could only be `void`.